src/Proto.Cluster/PidCache.cs

```diff
@@ -37,9 +40,12 @@ internal void Stop()
 
         private void OnMemberStatusEvent(MemberStatusEvent evn)
         {
-            if (evn is MemberLeftEvent || evn is MemberRejoinedEvent)
+            switch (evn)
             {
-                RemoveCacheByMemberAddress(evn.Address);
+                case MemberLeftEvent _:
+                case MemberRejoinedEvent _:
+                    RemoveCacheByMemberAddress(evn.Address);

```

Comment:
The means a full clear of pid cache on both join and leave events. right?

Reply:
That's right because partitions change in both cases.